### ISSUE TITLE
ORCID: update deposit edit form to allow users to enter ORCIDs for individual authors and contributors

### ISSUE THREAD

**Issue by amyehodge** [sul-dlss/happy-heron#1734]
Update the Release and Visibility section of the form to allow entry of ORCID and retrieval of a name OR manual entry of names without ORCIDs.

Mockups: https://projects.invisionapp.com/share/EQXC9CLKCR2#/screens/455996975_orcid

There are 3 changes to what is shown in the mockups:
1. Text next to the checkbox should read "ORCID iD (name will automatically be filled in)” and NOT "Use ORCID iD to enter this information"
2. No verification of the entry will be required. In other words, from IIIa, once the ORCID is entered and searched there will be no overlay box but instead will skip to IV but include the option to "Search again" that returns the user to the empty ORCID entry field.
3. There will be no overlay field in IIIb. Instead, the text that is shown in the overlay box will display directly on the page.

**Comment by jcoyne** [comment]
@astridu can you expand this design to show what happens when the user hits the search button?  What is the process if they realize they have the wrong orcid?  What if the orcid is not in the right format? How do you confirm this is the correct person?

**Comment by astridu** [comment]
1. When user hits the search, can we show a spinning icon to indicate we are retrieving data?
![image](https://user-images.githubusercontent.com/42216982/125139026-9ef4fe80-e0c4-11eb-9029-3eeec9abef5b.png)
2. If they have the wrong ORCID, they have to re-enter.
3. Give them an error. Can we have different error messages for different error states?
If error is user entered incorrect id, message is "Invalid ORCID iD."
If error is on our side or API is down, message is "Unable to retrieve ORCID iD. Try again later."
![image](https://user-images.githubusercontent.com/42216982/125139165-e7acb780-e0c4-11eb-818f-d18298c896aa.png)

4. I think we're assuming user knows their ORCID iD and that even if they have the same name as someone else, their iD is unique.

**Comment by amyehodge** [comment]
RE: confirming it's the right person. Sometimes a user will be entering ORCIDs for other authors (sometimes many other authors), but typically they will get the ORCIDs from those people. Many people don't have anything public in their ORCID records, so getting the number from the person is the best way to know you have the right person. As long as the name returned matches what they are expecting, I think it's good. They are highly unlikely to make a typo and pull up another person with the same name.

**Comment by jcoyne** [comment]
@astridu can you update the designs so that after the user does a lookup the orcid and the control are no longer useable. We don't want people to make changes after they make a selection.

**Comment by astridu** [comment]
Do you mean the radio button selection would go away? But the ORCID lookup can't be removed because what if they entered an incorrect number and ended up with someone else's ORCID?
![image](https://user-images.githubusercontent.com/42216982/125346538-b9221d00-e30e-11eb-906d-1723208cb976.png)

**Comment by jcoyne** [comment]
@astridu if they put in the wrong ORCID, then they could delete the row and add a new one.  That way we can keep ORCID and name in the same column saving space.  We only need to ask the "ORCID available?" question when they want to add a new identity. Otherwise we can just show "Maya Aguirre (ORCID: 0000-1111-2222-3333)"

**Comment by amyehodge** [comment]
@jcoyne @jcoyne regarding the discussion earlier today on editing names we retrieve from ORCID, I contacted Sheila Rabun at ORCID/Lyrasis regarding this question. My message and her response are below. 

TL;DR Editing is fine, assuming it's not causing us other problems like what @jcoyne brought up at the meeting. I do not want editing to require an admin to approve (yuck). We do not plan to have ORCIDs retrieved from DOI metadata as she mentioned. Someone may want to check the recommended workflow for retrieving IDs that she points to at [ORCID & Repositories](https://lyrasisnow.org/orcid-repositories/).

Apologies to @justinlittman and others if I was too strident in my opposition to this. I'll only say I had not had a great morning.

Hi Sheila,

We are finally (finally!) working on ORCID integration for our repository self-deposit system. I’m the product owner for the project. We were having a discussion about the UI at our planning meeting today. Our intent is to allow users to enter a DOI for an author and then retrieve the name for that person back into the form to store in our metadata.

The question came up as to whether we should allow editing of the name once it is retrieved. I was wondering if there were best practices from ORCID around doing that. There may be cases where a person wants to be referred to differently than what is on their ORCID record. As an example, I had a case today (not related to ORCID), where some authors wanted the names on their data deposit to match how they were listed on the publication that the data was related to.

There may be some other issues with allowing editing (like, what if they change the name to someone else’s name entirely and now the name and the identifier don’t match at all?), but I’m wondering if there is any guidance from ORCID around editing names once they are retrieved from ORCID.

Thanks!
Amy Hodge


Hi Amy,

Thanks for contacting me. That’s great to hear! I have a few insights:

-If the name retrieval is coming via the ORCID API/ORCID record, ideally the individual author(s) would have their primary name in ORCID appear the way they want to be credited. ORCID provides a few different name fields for this reason: [Add and edit your name on your ORCID record – ORCID](https://support.orcid.org/hc/en-us/articles/360006973853-Add-and-edit-your-name-on-your-ORCID-record) - so ideally each author would have their “published name” the way they want it - they can control the way their name appears in ORCID, and then that will subsequently populate other systems that are integrated to pull their name from ORCID. However, if an author wants their name to appear differently in the IR for whatever reason, it’s fine to change the name after it has been imported from ORCID. To prevent mis-use, you could make it so that the individual can’t change the name, but can request to have it changed by an admin.

-If you are having people enter a DOI to retrieve author names, depending on whether the ORCID iDs were entered in the DOI metadata, sometimes the metadata of the DOI is different from metadata on an ORCID record, so that is something to be aware of.

-If someone is submitting an item to the IR that has multiple authors and you want to get the ORCID iDs for each author, the individual who is submitting the item can connect their own ORCID iD via the API, but there is a recommended workflow for getting authenticated ORCID iDs for co-authors too - described in the Dryad section of this blog: [ORCID & Repositories](https://lyrasisnow.org/orcid-repositories/)

Don’t hesitate to let me know if there is anything else I can help with as you move forward.

best,
Sheila

**Comment by astridu** [comment]
So then if there is only 1 author listed, that entry will need a trash icon, right? 

![image](https://user-images.githubusercontent.com/42216982/125364875-1b3b4c00-e328-11eb-8447-f61b0864b70f.png)

**Comment by amyehodge** [comment]
@astridu Yes, I think that's true. That brings us back to the reason the first author doesn't have a trash can icon -- we didn't want to completely delete that entry row because then there are no boxes there and it was a little confusing to figure out what to do without boxes. The user needs to click on the "Add another author" link, but it seems like this is going to have to be deletable. @jcoyne do you see any other solution?

**Comment by amyehodge** [comment]
@astridu Can we also delete the "is" from these entry rows? @justinlittman pointed out that these now read somewhat confusingly "[Role term] is [Is ORCID available?]...")

@jcoyne Does @astridu's last update work for you?

**Comment by astridu** [comment]
Also put 'clear values' link instead of trash can for the first entry. Clicking this link would clear the values entered so the user can re-enter if the ORCID isn't right for some reason.

![image](https://user-images.githubusercontent.com/42216982/125843607-85d703a6-e81e-4199-abab-e254d28619d1.png)

**Comment by jcoyne** [comment]
@astridu another concern that came up is that typicially when the choice is yes/no (or true/false), a checkbox is used.  Is there a reason we wouldn't want to have:

- [ ] Lookup with ORCiD?

**Comment by astridu** [comment]
That could work too. Would the default be to have it selected?
![image](https://user-images.githubusercontent.com/42216982/125865994-d6555429-67be-4df0-86fc-a529b0efa7c1.png)


And then when it's not selected, it would display the first/last name text fields.
![image](https://user-images.githubusercontent.com/42216982/125866032-e9f5bbc3-58d8-4dfd-afcd-c04d28f16f57.png)

**Comment by justinlittman** [comment]
@astridu Should the user be able to change the value returned by ORCID?

**Comment by justinlittman** [comment]
Also, what does "Lookup ORCID id" mean? Naively, I would think that means given a name, find the ORCID id, not the other way around.

**Comment by astridu** [comment]
@justinlittman User should not be able to change the value returned by ORCID. 
@jcoyne suggested 'Lookup ORCID iD' What would make more sense to you? "Use ORCID id" or "Enter ORCID iD"?

**Comment by amyehodge** [comment]
1. Questions till remaining:
2. Select alternate text if we don't like "Lookup ORCID iD"
3. Decide what information to return from the ORCID iD lookup. If we can get name and institution, that would be great. In some cases we can only get name. In some cases we won't even be able to see the name. When there isn't even a name, we should display a message like, "No public information available for this record." They should still be able to select this ID, because they may know for sure that it's the right one.
4. I prefer the "clear values" link to the trash can, but if others find the consistency of the trash can better, that's fine with me.
5. There is room next to the editable name fields for some explanation, and I think it needs some. Perhaps "You may modify how this name is displayed for this work."

**Comment by ndushay** [comment]
Per mid-sprint planning:  
- we need better UX design on individual contributor names <--> ORCID lookups and name from ORCID edits
- JCoyne says it will be a lot faster to only implement this once (with an acceptable UX design)
- Astrid is on vacation all week
- Vivian will ask Mark if Gary can give us a few hours this week
